designs/disruption-controls-by-action.md

```diff
@@ -0,0 +1,218 @@
+}
+```
+
+#### Example 
+
+```yaml 
+apiVersion: karpenter.sh/v1beta1
+kind: NodePool
+metadata:
+  name: example-nodepool
+spec:
+  disruption:
+    consolidationPolicy: WhenUnderutilized
+    genericPerActionControls:
+      Consolidation:
+        disruptAfter: "30m"
+        budgets:
+          - nodes: "20%"
+            schedule: "0 0 1 * *"
+            duration: "1h"
+      Drift:
+        disruptAfter: "1h"
+        budgets:
+          - nodes: "10%"
+            schedule: "0 0 * * 0"
+            duration: "2h"
+      Expiration:
+        disruptAfter: "Never"
+```
+
+#### Considerations 
+Some of the API choices for a given action seem to follow a similar pattern. These include ConsolidateAfter, ExpireAfter, and there are discussions about introducing a global DisruptAfter. Moreover, when discussing disruption budgets, we talk about adding behavior for each action. It appears there is a need for disruption controls within the budgets for each action, not just overall.
+
+This approach aligns well with controls that apply to all existing actions. The proposal presented here is similar to the one mentioned above in relation to the actions we allow to be defined (All, Consolidation, Drift, Expiration, Emptiness).
+
+This proposal is currently scoped for disruptionBudgets by action. However, we should also consider incorporating other generic disruption controls into the PerActionControls, even if we do not implement them immediately. Moving ConsolidateAfter and ExpireAfter into the per-action controls is a significant migration that requires careful planning and its own dedicated design. This proposal simply demonstrates a potential model that highlights the benefits of defining controls at a per-action level of granularity.
+
+### Pros + Cons 
+* 👍 This model starts to make more sense as we continue to add general behaviors that apply to all disruption actions where users will want control on the action level.
+* 👍 Provides place per action for generic controls intended to be shared across all disruption methods. While not all methods share the same disruption actions, there are already cases for this. 
+* 👍 Could extend other fields beyond generic values for specific actions with validation   
+*  👎👎 Breaking API Change for Budgets at least, and if we decide to model DisruptAfter we also would have to break those apis. It might make sense to break the budgets now before they have garnered large adoption as it becomes harder to make the change as time goes on.
+* 👎 Doesn't allow for easy defaulting of `All` disruption actions
+* 👎 Adds complexity to the use of budgets, before its high level on the disruption controls but with this design approach its nested inside another field.
+
+
+### Conclusion: Preferred Design
+If the goal is to provide a simple, backward-compatible solution with immediate applicability, Approach A is more suitable. It provides a straightforward way to manage disruptions without overhauling the existing system.
```

Comment:
Suggestion: Generally, we end our designs with a recommendation. We can discuss more about what we as the maintainers think is a good idea here, but I'd come in with your own conviction about what you think the right direction is here and then we can discuss if we agree/disagree.

Reply:
yes i agree, doc was still in draft phase but I think im going to pick out option A. Option b is too much change even though its a fun idea its completely breaking